manager-rs/dauth-service/src/tasks/replace_key_shares.rs

```diff
@@ -0,0 +1,68 @@
+use std::sync::Arc;
+
+use crate::data::context::DauthContext;
+use crate::data::error::DauthError;
+use crate::database;
+use crate::database::tasks::replace_key_shares::ReplaceKeyShareTask;
+use crate::rpc::clients;
+
+/// Runs the replace key shares task.
+/// If any replacements are queued up, waits for 10 seconds and then calls
+/// the key share replacement rpc.
+pub async fn run_task(context: Arc<DauthContext>) -> Result<(), DauthError> {
+    let mut transaction = context.local_context.database_pool.begin().await.unwrap();
+    let replaces = database::tasks::replace_key_shares::get(&mut transaction).await?;
+    transaction.commit().await.unwrap();
+
+    if replaces.is_empty() {
+        tracing::info!("Nothing to do for replace key share task");
+    } else {
+        tracing::info!("Found {} replace key share(s) pending", replaces.len());
+
+        let mut tasks = Vec::new();
+
+        for replace in replaces {
+            tasks.push(tokio::spawn(replace_key_share(context.clone(), replace)));
+        }
+
+        for task in tasks {
+            match task.await {
```

Comment:
A different pattern you could use is instead of waiting on the join handle, spawn the task with no handle, but pass it a [MPSC channel](https://docs.rs/tokio/latest/tokio/sync/mpsc/fn.channel.html) where the task can report its status before terminating. Then your main run function can select on both the channel receive (for status from any of the many outstanding tasks) or a tick (for a periodic polling timeout).

Alternatively we could enable that tokio unstable feature to wait on a vector of futures. I don't think that would require moving to nightly rust, we would just open ourselves up to API breakage if we update the version of tokio we're using.

Reply:
The goal I had in mind is to run each subtask at the same time, since these will spend the majority of time waiting on db/network calls. run_task should block once everything starts running, and wait for them all to finish (but they will continue to run even without the join). The join at the end will ensure everything finishes before returning , and check join status. 

Even though this is joining the first task, it shouldn't matter since all of them are running in the background. I tested this to be sure it behaves as I expected. Is there something else you had in mind?